#3027 Added configurable length of point names in reports

Patrykb0802 opened the pull request.

Patrykb0802 committed:
#3027 Added configurable length of point names in reports

Limraj commented on src/org/scada_lts/utils/SystemSettingsUtils.java:
```diff
@@ -74,6 +74,7 @@ private SystemSettingsUtils() {}
     private static final String SECURITY_HTTP_QUERY_PROTECT_ENABLED_KEY = "scadalts.security.http.query.protect.enabled";
 
     private static final String CUSTOM_CSS_CONTENT_KEY = "systemsettings.custom.css.content";
+    private static final String POINT_NAMES_LENGTH_IN_REPORT_KEY = "systemsettings.point.names.length.in.reports";
```
systemsettings.reports.data-point-extended-name-length.limit

Limraj commented on src/com/serotonin/mango/vo/report/ReportChartCreator.java:
```diff
@@ -147,8 +148,9 @@ else if (pointStat.getDiscreteTimeSeries() != null)
+            SystemSettingsService systemSettingsService = new SystemSettingsService();
             int consolidatedChartHeight = ImageChartUtils.calculateHeightChart(pointStatistics, IMAGE_HEIGHT_PIXELS,
-                    POINT_LABEL_HEIGHT_IN_LEGEND_PIXELS, LINE_LENGTH_IN_LEGEND_LIMIT, DATA_POINT_EXTENDED_NAME_LENGTH_LIMIT);
+                    POINT_LABEL_HEIGHT_IN_LEGEND_PIXELS, LINE_LENGTH_IN_LEGEND_LIMIT, systemSettingsService.getPointNamesLengthInReport());
```
We change all parameter variable names according to this pattern: DATA_POINT_EXTENDED_NAME_LENGTH_LIMIT 

If you move a variable to the configuration, leave the name of this parameter, adding only the context, i.e. Report.

systemSettingsService.getDataPointExtendendNameLengthInReportLimit()
systemSettingsService.setDataPointExtendendNameLengthInReportLimit()

Limraj commented on src/com/serotonin/mango/web/dwr/SystemSettingsDwr.java:
```diff
@@ -394,6 +396,12 @@ public DwrResponseI18n saveMiscSettings(int uiPerformance, String dataPointRtVal
+		}
+		else{
+			systemSettingsDAO.setIntValue(SystemSettingsDAO.POINT_NAMES_LENGTH_IN_REPORT, pointNamesLengthInReport);
+		}
```
use:
`systemSettingsService.setDataPointExtendendNameLengthInReportLimit(dataPointExtendendNameLengthLimit);`
```
setDataPointExtendendNameLengthInReportLimit(dataPointExtendendNameLengthLimit) {
systemSettingsDAO.setIntValue(SystemSettingsDAO.corrected_name, dataPointExtendendNameLengthLimit);
}
```

Patrykb0802 committed:
#3027 Added configurable length of point names in reports

Patrykb0802 committed:
#3027 Added configurable length of point names in reports

Limraj commented on src/org/scada_lts/utils/SystemSettingsUtils.java:
```diff
@@ -74,6 +74,7 @@ private SystemSettingsUtils() {}
     private static final String SECURITY_HTTP_QUERY_PROTECT_ENABLED_KEY = "scadalts.security.http.query.protect.enabled";
 
     private static final String CUSTOM_CSS_CONTENT_KEY = "systemsettings.custom.css.content";
+    private static final String DATA_POINT_EXTENDED_NAME_LENGTH_IN_REPORTS_LIMIT = "systemsettings.reports.data-point-extended-name-length.limit";
```
You add this as the last field because the last one was added. 
You do the same with the method, it should be at the end of other public methods that were previously added.

`systemsettings.reports.data-point-extended-name-length.limit=64`
I don't see this key in env.properties?

Limraj commented on src/org/scada_lts/utils/SystemSettingsUtils.java:
```diff
@@ -380,6 +381,16 @@ public static int getThreadsNameAdditionalLength() {
         }
     }
 
+    public static int getDataPointExtendedNameLengthInReportsLimit() {
```
It should be at the end of other public methods that were previously added.

Limraj commented on src/com/serotonin/mango/vo/report/ReportChartCreator.java:
```diff
@@ -147,8 +147,9 @@ else if (pointStat.getDiscreteTimeSeries() != null)
                 // The path comes from the servlet path definition in web.xml.
                 model.put("chartName", IMAGE_SERVLET + chartName);
             }
+            SystemSettingsService systemSettingsService = new SystemSettingsService();
```
Create an instance field with this service in this class instead of making new everywhere.

Limraj commented on src/org/scada_lts/utils/SystemSettingsUtils.java:
```diff
@@ -380,6 +381,16 @@ public static int getThreadsNameAdditionalLength() {
 
+    public static int getDataPointExtendedNameLengthInReportsLimit() {
+        try {
+            String config = ScadaConfig.getInstance().getConf().getProperty(DATA_POINT_EXTENDED_NAME_LENGTH_IN_REPORTS_LIMIT, "64");
```
In env.properties you need to set 64, and here you can set 65 to know what the source of the default value is and whether something went wrong while reading the properties from the file.

Limraj commented on src/org/scada_lts/utils/SystemSettingsUtils.java:
```diff
@@ -380,6 +381,16 @@ public static int getThreadsNameAdditionalLength() {
+            return Integer.parseInt(config);
+        } catch (Exception e) {
+            LOG.error(e.getMessage());
+            return 64;
```
65 or 66

Limraj commented on webapp-resources/messages_en.properties:
```diff
@@ -3368,3 +3368,4 @@ header.imageSets=Image sets
 header.dynamicImages=Dynamic images
 systemSettings.customCssSaved=Custom Css has been saved
 systemSettings.invalidCustomCss=Invalid custom CSS
+systemSettings.pointNamesLengthInGeneratedReports= Length of data points names in generated reports
```
-> systemsettings.reports.dataPointExtendedNameLengthLimit=Point name length limit in reports

And:
**The key must be moved to the file with each language.**

Limraj commented on src/com/serotonin/mango/web/dwr/SystemSettingsDwr.java:
```diff
@@ -394,7 +397,12 @@ public DwrResponseI18n saveMiscSettings(int uiPerformance, String dataPointRtVal
+		if(pointExtendedNameLengthInReportsLimit < 0) {
+			response.addContextualMessage(SystemSettingsDAO.DATA_POINT_EXTENDED_NAME_LENGTH_IN_REPORTS_LIMIT, "validate.invalidValue");
+		}
+		else{
```
`} else { ` 

The above also needs to be improved.

Use:
```
if (condition) {
...
} else {
...
}
```

Patrykb0802 committed:
#3027 Added configurable length of point names in reports

Limraj committed:
Merge remote-tracking branch 'origin/release/2.8.0' into feature/#3027_Added_configurable_length_of_point_names_in_reports

Limraj committed:
#3027 Added configurable length of point names in reports:
- Refactored method SystemSettingsDwr.saveMiscSettings moved logic to SystemSettingsService;
- Added static method ImageChartUtils.truncatePointNameForReport(String, SystemSettingsService);
